test/csuite/wt6616_checkpoint_oldest_ts/main.c

```diff
@@ -0,0 +1,372 @@
+
+    WT_UNUSED(sig);
+    pid = wait(NULL);
+    /* The core file will indicate why the child exited. Choose EINVAL here. */
+    testutil_die(EINVAL, "Child process %" PRIu64 " abnormally exited", (uint64_t)pid);
+}
+
+extern int __wt_optind;
+extern char *__wt_optarg;
+
+int
+main(int argc, char *argv[])
+{
+    struct sigaction sa;
+    struct stat sb;
+    WT_CONNECTION *conn;
+    WT_CURSOR *cursor;
+    WT_DECL_RET;
+    WT_RAND_STATE rnd;
+    WT_SESSION *session;
+    pid_t pid;
+    uint64_t oldest_ts, stable_ts, ts;
+    uint32_t timeout;
+    int ch, status;
+    char kname[64], statname[1024], tscfg[64];
+    char ts_string[WT_TS_HEX_STRING_SIZE];
+    const char *working_dir;
+    bool fatal, rand_time;
+
+    (void)testutil_set_progname(argv);
+
+    rand_time = false;
+    timeout = MIN_TIME;
+    working_dir = "WT_TEST.wt6616-checkpoint-oldest-ts";
+
+    while ((ch = __wt_getopt(progname, argc, argv, "h:lt:")) != EOF)
+        switch (ch) {
+        case 'h':
+            working_dir = __wt_optarg;
+            break;
+        case 'l':
+            log_table = true;
+            break;
+        case 't':
+            rand_time = false;
+            timeout = (uint32_t)atoi(__wt_optarg);
+            break;
+        default:
+            usage();
+        }
+    argc -= __wt_optind;
+    if (argc != 0)
+        usage();
+
+    testutil_work_dir_from_path(home, sizeof(home), working_dir);
+    testutil_make_work_dir(home);
+    if (rand_time) {
+        timeout = __wt_random(&rnd) % MAX_TIME;
+        if (timeout < MIN_TIME)
+            timeout = MIN_TIME;
+    }
+
+    printf("CONFIG: %s -h %s -t %" PRIu32 " %s\n", progname, working_dir, timeout,
+      log_table ? "-l" : "");
+    /*
+     * Fork a child to insert and delete as many items. We will then randomly kill the child, run
+     * recovery and make sure all items from the oldest to stable timestamps of the checkpoint exist
+     * after recovery runs.
+     */
+    memset(&sa, 0, sizeof(sa));
+    sa.sa_handler = handler;
+    testutil_checksys(sigaction(SIGCHLD, &sa, NULL));
+    testutil_checksys((pid = fork()) < 0);
+
+    if (pid == 0) { /* child */
+        run_workload();
+        return (EXIT_SUCCESS);
+    }
+
+    /* parent */
+    /*
+     * Sleep for the configured amount of time before killing the child. Start the timeout from the
+     * time we notice that the file has been created. That allows the test to run correctly on
+     * really slow machines.
+     */
+    testutil_check(__wt_snprintf(statname, sizeof(statname), "%s/%s", home, ckpt_file));
+    while (stat(statname, &sb) != 0)
+        testutil_sleep_wait(1, pid);
+    sleep(timeout);
+    sa.sa_handler = SIG_DFL;
+    testutil_checksys(sigaction(SIGCHLD, &sa, NULL));
+
+    printf("Kill child\n");
+    testutil_checksys(kill(pid, SIGKILL) != 0);
+    testutil_checksys(waitpid(pid, &status, 0) == -1);
+
+    if (chdir(home) != 0)
+        testutil_die(errno, "parent chdir: %s", home);
+
```

Comment:
I recommend re-instating the code to make a copy of the original database before recovery. It can be helpful looking at bugs (and that also means eventually you'll likely want the `-v` verify-only arg).